tests/unit/js/test.search.js

```diff
@@ -0,0 +1,182 @@
+}
+
+/** Runs visibility tests for all 'facetEntry' elements in document.
+ *
+ * @param {Number} totalFacet           total number of facet
+ * @param {Number} expectedVisibleFacet expected number of visible facet
+ */
+function checkFacetVisibility(totalFacet, expectedVisibleFacet) {
+    const facetEntryList = document.getElementsByClassName('facetEntry');
+
+    test('facetEntry element number', () => {
+        expect(facetEntryList).toHaveLength(totalFacet);
+    });
+
+    for (let i = 0; i < totalFacet; i++) {
+        if (i < expectedVisibleFacet) {
+            test(`element "facet_${i+1}" displayed`, () => {
+                expect(facetEntryList[i].classList.contains('ui-helper-hidden')).toBe(false);
+            });
+        } else {
+            test(`element "facet_${i+1}" hidden`, () => {
+                expect(facetEntryList[i].classList.contains('ui-helper-hidden')).toBe(true);
+            });
+        }
+    }
+}
+
+/** Runs visibility tests for 'less', 'bull' and 'more' elements in document
+ *
+ * @param {Number} totalFacet           total number of facet
+ * @param {Number} minVisibleFacet      minimum visible facet number
+ * @param {Number} expectedVisibleFacet expected number of visible facet
+ */
+function checkFacetMoreLessVisibility(totalFacet, minVisibleFacet, expectedVisibleFacet) {
+    if (expectedVisibleFacet <= minVisibleFacet) {
+        test('element "test_more"', () => {
+            expect(document.getElementById('test_more').style.display).not.toBe('none');
+        });
+        test('element "test_bull"', () => {
+            expect(document.getElementById('test_bull').style.display).toBe('none');
+        });
+        test('element "test_less"', () => {
+            expect(document.getElementById('test_less').style.display).toBe('none');
+        });
+    } else if (expectedVisibleFacet >= totalFacet) {
+        test('element "test_more"', () => {
+            expect(document.getElementById('test_more').style.display).toBe('none');
+        });
+        test('element "test_bull"', () => {
+            expect(document.getElementById('test_bull').style.display).toBe('none');
+        });
+        test('element "test_less"', () => {
+            expect(document.getElementById('test_less').style.display).not.toBe('none');
+        });
+    } else {
+        test('element "test_more"', () => {
+            expect(document.getElementById('test_more').style.display).not.toBe('none');
+        });
+        test('element "test_bull"', () => {
+            expect(document.getElementById('test_bull').style.display).not.toBe('none');
+        });
+        test('element "test_less"', () => {
+            expect(document.getElementById('test_less').style.display).not.toBe('none');
+        });
+    }
+}
+
+beforeAll(() => {
+    global.$ = jquery;
+
+    // Replace getClientRects to enable jQuery ':hidden' selector used by 'more' and 'less' functions
+    window.Element.prototype.getClientRects = function() {
```

Comment:
When stubbing global properties, it's important to teardown to restore things to how they were. This way you won't introduce side effects to other tests that might be using this global code.

```
var _original = window.Element.prototype.getClientRects ;
afterEach(() => {
  window.Element.prototype.getClientRects = _original 
});```